### PR TITLE
doc: revisions info adjustments

### PR DESCRIPTION
**Pull request by annwoj**
edited versions and revisions section.
related to Jira NCSDK20957.

### CODE REVIEW

File: doc/nrf/developing/code_base.rst [hunk 1]
```diff
@@ -68,33 +68,60 @@ More information about manifests can be found in the :ref:`west manifest section
...
+Versions and revisions
+**********************
+
+The |NCS| version numbers are assigned based on very specific criteria.
+They follow the ``MAJOR.MINOR.PATCH`` pattern (also further called as ``X.Y.Z``), with possible additional postfixes depending on the release.
+
++-------------------------------+----------------------------------------------------------------------------+--------------------------------------------------------------------------------------------+
+| Release type                  | Release critera                                                            | Comments                                                                                   |
++===============================+============================================================================+============================================================================================+
+| ``MAJOR``                     | Introduces a large number of substantial changes across the board          | Version number is increased rarely.                                                        |
```
Comment (greg-fer) [inline doc/nrf/developing/code_base.rst:80]:
```suggestion
| ``MAJOR``                     | Introduces a large number of substantial changes across the board.          | Major version number is increased rarely and the release is accompanied by a migration guide.                                                        |
```

File: doc/nrf/developing/code_base.rst [hunk 2]
```diff
@@ -68,33 +68,60 @@ More information about manifests can be found in the :ref:`west manifest section
...
+| ``MINOR``                     | Introduces a new functionality.                                            | Version number is increased every time a major release is cut.                             |
+|                               |                                                                            | Minor releases are the default types of an |NCS| release.                                  |
+|                               |                                                                            | They introduce new functionalities and may break APIs.                                     |
++-------------------------------+----------------------------------------------------------------------------+--------------------------------------------------------------------------------------------+
+| ``PATCH``                     | Introduces a minor or a bugfix release.                                    | Patch releases only address functional issues but do not introduce a new functionality.    |
```
Comment (greg-fer) [inline doc/nrf/developing/code_base.rst:86]:
```suggestion
| ``PATCH``                     | Introduces minor changes to existing functionalities or bugfixes.                                    | Patch releases only address functional issues and do not introduce new functionalities.    |
```

**Comment by greg-fer** [comment]
Actually, I've also spotted that this section could be updated with a link to the new section:

https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/introduction.html#documentation-pages

Old text:

> To access different versions of the nRF Connect SDK documentation, use the version drop-down in the top right corner. A “99” at the end of the version number of this documentation indicates continuous updates on the main branch since the previous major.minor release.

Proposal:

> To access different versions of the nRF Connect SDK documentation, use the version drop-down in the top right corner. The documentation versions follow the :ref:`dm-revisions` of the SDK.

File: doc/nrf/developing/code_base.rst (hunk 2 above)
Comment (annwoj) [reply doc/nrf/developing/code_base.rst]: rephrased this part a bit

**Comment by greg-fer** [comment]
@shanthanordic , @nordicjm , could you have a look at this PR? Thanks!

**Review by nordicjm — APPROVED** [review]
Looks OK

**Pull request merged**